server/src/main/kotlin/suwayomi/tachidesk/graphql/mutations/IReaderSourceMutation.kt

```diff
@@ -0,0 +1,186 @@
+import suwayomi.tachidesk.graphql.types.IReaderChapterType
+import suwayomi.tachidesk.graphql.types.IReaderNovelType
+import suwayomi.tachidesk.graphql.types.IReaderPageType
+import suwayomi.tachidesk.manga.impl.IReaderNovel
+import suwayomi.tachidesk.manga.model.table.ChapterTable
+import suwayomi.tachidesk.manga.model.table.IReaderNovelTable
+import suwayomi.tachidesk.manga.model.table.MangaTable
+import suwayomi.tachidesk.server.JavalinSetup.future
+import java.util.concurrent.CompletableFuture
+
+class IReaderSourceMutation {
+    enum class FetchIReaderNovelType {
+        SEARCH,
+        POPULAR,
+        LATEST,
+    }
+
+    data class FetchIReaderNovelsInput(
+        val clientMutationId: String? = null,
+        @GraphQLDescription("Source ID")
+        val source: Long,
+        @GraphQLDescription("Type of fetch operation")
+        val type: FetchIReaderNovelType,
+        @GraphQLDescription("Page number (1-indexed)")
+        val page: Int,
+        @GraphQLDescription("Search query (required for SEARCH type)")
+        val query: String? = null,
+    )
+
+    data class FetchIReaderNovelsPayload(
+        val clientMutationId: String?,
+        val novels: List<IReaderNovelType>,
+        val hasNextPage: Boolean,
+    )
+
+    @RequireAuth
+    @GraphQLDescription("Fetch novels from an IReader source")
+    fun fetchIReaderNovels(input: FetchIReaderNovelsInput): CompletableFuture<DataFetcherResult<FetchIReaderNovelsPayload?>> {
+        val (clientMutationId, sourceId, type, page, query) = input
+
+        return future {
+            asDataFetcherResult {
+                require(page > 0) { "Page must be greater than 0" }
+
+                val novelsPage =
+                    when (type) {
+                        FetchIReaderNovelType.SEARCH -> {
+                            require(!query.isNullOrBlank()) { "Query is required for SEARCH type" }
+                            IReaderNovel.searchNovels(sourceId, query, page)
+                        }
+                        FetchIReaderNovelType.POPULAR -> {
+                            IReaderNovel.getPopularNovels(sourceId, page)
+                        }
+                        FetchIReaderNovelType.LATEST -> {
+                            IReaderNovel.getLatestNovels(sourceId, page)
+                        }
+                    }
+
+                FetchIReaderNovelsPayload(
+                    clientMutationId = clientMutationId,
+                    novels = novelsPage.mangas.map { IReaderNovelType(it, sourceId) },
+                    hasNextPage = novelsPage.hasNextPage,
+                )
+            }
+        }
+    }
+
+    data class FetchIReaderChaptersInput(
+        val clientMutationId: String? = null,
+        @GraphQLDescription("Source ID (required if using novelUrl)")
+        val source: Long? = null,
+        @GraphQLDescription("Novel URL/key from the source (provide this OR novelId)")
+        val novelUrl: String? = null,
```

Comment:
This and the sourceId should be removed, at this point we should be relying only on the novelId. Use that to get the sourceId.